### PR TITLE
Close all Response in samples

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
Part of https://github.com/square/okhttp/issues/3132

File: samples/crawler/src/main/java/okhttp3/sample/Crawler.java [hunk 1]
```diff
@@ -95,32 +95,35 @@ public void fetch(HttpUrl url) throws IOException {
...
+          + " over "
+          + response.protocol()
+          + ")") : "(cache)";
+      int responseCode = response.code();
+
+      System.out.printf("%03d: %s %s%n", responseCode, url, responseSource);
+
+      String contentType = response.header("Content-Type");
+      if (responseCode != 200 || contentType == null) {
+        response.body().close();
```
Comment (swankjesse) [inline samples/crawler/src/main/java/okhttp3/sample/Crawler.java]: Unnecessary now!

File: samples/crawler/src/main/java/okhttp3/sample/Crawler.java [hunk 2]
```diff
@@ -95,32 +95,35 @@ public void fetch(HttpUrl url) throws IOException {
...
+      }
 
-    MediaType mediaType = MediaType.parse(contentType);
-    if (mediaType == null || !mediaType.subtype().equalsIgnoreCase("html")) {
-      response.body().close();
-      return;
-    }
+      MediaType mediaType = MediaType.parse(contentType);
+      if (mediaType == null || !mediaType.subtype().equalsIgnoreCase("html")) {
+        response.body().close();
```
Comment (swankjesse) [inline samples/crawler/src/main/java/okhttp3/sample/Crawler.java]: Same.

File: samples/simple-client/src/main/java/okhttp3/sample/OkHttpContributors.java [hunk 3]
```diff
@@ -31,23 +31,23 @@ public static void main(String... args) throws Exception {
...
-
-    // Sort list by the most contributions.
-    Collections.sort(contributors, new Comparator<Contributor>() {
-      @Override public int compare(Contributor c1, Contributor c2) {
-        return c2.contributions - c1.contributions;
+    try (Response response = client.newCall(request).execute()) {
+      // Deserialize HTTP response to concrete type.
+      ResponseBody body = response.body();
+      List<Contributor> contributors = CONTRIBUTORS_JSON_ADAPTER.fromJson(body.source());
+      body.close();
```
Comment (swankjesse) [inline samples/simple-client/src/main/java/okhttp3/sample/OkHttpContributors.java]: Same.

**Pull request merged**